docs/source/image_process.mdx

```diff
@@ -0,0 +1,127 @@
+>>> dataset[0]["image"]
+```
+
+![image_process_beignet](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/image_process_beignet.png)
+
+
+To load an image from its path, use the [`~Dataset.cast_column`] method. The [`~Image`] feature will decode the data at the path to return an image object:
+
+```py
+>>> from datasets import load_dataset, Image
+
+>>> dataset = load_dataset("cats_vs_dogs").cast_column("image_file_path", Image())
+>>> dataset[0]["image_file_path"]
+```
+
+![image_process_dog](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/image_process_dog.png)
+
+<Tip>
+
+You can also access the path and bytes of an image file by setting `decode=False` when you load a dataset. In this case, you will need to cast the `image` column:
+
+`dataset = load_dataset("food101", split="train[100:]").cast_column('image', Image(decode=False))`
+
+</Tip>
+
+## ImageFolder
+
+You can also load your image dataset with our `ImageFolder` dataset builder without writing a custom dataloader. Your image dataset structure should look like this:
+
+```
+folder/dog/golden_retriever.png
+folder/dog/german_shepherd.png
+folder/dog/chihuahua.png
+
+folder/cat/maine_coon.png
+folder/cat/bengal.png
+folder/cat/birman.png
+```
+
+Then you can load your dataset by specifying `imagefolder` and the directory of your dataset in `data_dir`:
+
+```py
+>>> from datasets import load_dataset
+>>> dataset = load_dataset("imagefolder", data_dir="/path/to/folder")
+```
+
+Load remote datasets from their URLs with the `data_files` parameter:
+
+```py
+dataset = load_dataset("imagefolder", data_files="https://download.microsoft.com/download/3/E/1/3E1C3F21-ECDB-4869-8368-6DEBA77B919F/kagglecatsanddogs_3367a.zip", split="train")
+```
+
+## Data augmentation
+
+Adding data augmentations to a dataset is common to prevent overfitting and achieve better performance. You can use any library or package you want to apply the augmentations. This guide will use the transforms from [torchvision](https://pytorch.org/vision/stable/transforms.html).
+
+Add the [`ColorJitter`](https://pytorch.org/vision/stable/transforms.html#torchvision.transforms.ColorJitter) transform to change the color properties of the image randomly:
+
+```py
+>>> from torchvision.transforms import Compose, ColorJitter, ToTensor
+
+>>> jitter = Compose(
+    [
+     ColorJitter(brightness=0.25, contrast=0.25, saturation=0.25, hue=0.7),
+     ToTensor(),
+    ]
+)
+```
+
+Create a function to apply the `ColorJitter` transform to an image:
+
+```py
+>>> def transforms(examples):
+...     examples["pixel_values"] = [jitter(image.convert("RGB")) for image in examples["image"]]
+...     return examples
+```
+
+Use the [`~Dataset.set_transform`] function to apply the transformation on-the-fly. This is generally faster than [`~Dataset.map`] because it only prepares the data when it is used, instead of preparing it across the entire dataset ahead of time:
```

Comment:
This is **slower** than `map` unless examples are accessed only once. However, `set_transform` consumes less memory because `map` generates a cache file the first time.

Reply:
"`set_transform ` consumes less memory" -> "`set_transform ` consumes less disk space" no ?

Reply:
Yes, less disk space. 🙂